internal/controllers/device_config_controller.go

```diff
@@ -1,165 +1,218 @@
+
+	// Query the database to get the PowerConfigs based on the template name using SQLBoiler
+	dbPowerConfig, err := models.PowerConfigs(
+		models.PowerConfigWhere.TemplateName.EQ(templateName),
+	).One(c.Context(), d.db)
+
+	// Error handling
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return fiber.NewError(fiber.StatusNotFound, "No Power Config data found for the given template name.")
+		}
+		return errors.Wrap(err, "Failed to retrieve Power Config")
+	}
+
+	apiPowerConfig := PowerConfig{
+
+		ID:                                     dbPowerConfig.ID,
+		Version:                                dbPowerConfig.Version,
+		BatteryCriticalLevelVoltage:            dbPowerConfig.BatteryCriticalLevelVoltage,
+		SafetyCutOutVoltage:                    dbPowerConfig.SafetyCutOutVoltage,
+		SleepTimerEventDrivenInterval:          dbPowerConfig.SleepTimerEventDrivenInterval,
+		SleepTimerEventDrivenPeriod:            dbPowerConfig.SleepTimerEventDrivenPeriod,
+		SleepTimerInactivityAfterSleepInterval: dbPowerConfig.SleepTimerInactivityAfterSleepInterval,
+		SleepTimerInactivityFallbackInterval:   dbPowerConfig.SleepTimerInactivityFallbackInterval,
+		WakeTriggerVoltageLevel:                dbPowerConfig.WakeTriggerVoltageLevel,
 	}
 
-	return c.JSON(powerConfig)
+	return c.JSON(apiPowerConfig)
+
 }
 
-// GetDBCFile godoc
-// @Description  Retrieve the URL pointing to the DBC file for a given VIN
+// GetDBCFileByTemplateName godoc
+// @Description  Fetches the DBC file from the dbc_files table given a template name
 // @Tags         vehicle-signal-decoding
-// @Produce      json
-// @Success      200 {string} string
-// @Param        vin  path   string  true   "vehicle identification number (VIN)"
-// @Router       /device-config/:vin/dbc [get]
-func (d *DeviceConfigController) GetDBCFile(c *fiber.Ctx) error {
-	baseURL := d.Settings.DeploymentURL
-	dbcURL := fmt.Sprintf("%s/default/dbc-config/%s.dbc", baseURL, c.Params("vin"))
-	return c.JSON(fiber.Map{"dbcFileUrl": dbcURL})
+// @Produce      plain
+// @Success      200 {string} string "Successfully retrieved DBC file"
+// @Failure 404 "No DBC file found for the given template name."
+// @Param        template_name  path   string  true   "template name"
+// @Router       /device-config/:template_name/dbc-file [get]
+func (d *DeviceConfigController) GetDBCFileByTemplateName(c *fiber.Ctx) error {
+	templateName := c.Params("template_name")
+
+	// Query the database using SQLBoiler
+	//use same logic as above
+	dbResult, err := models.DBCFiles(
+		models.DBCFileWhere.TemplateName.EQ(templateName)).One(c.Context(), d.db)
+
+	// Error handling
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return c.Status(fiber.StatusNotFound).SendString(fmt.Sprintf("No DBC file found for template name: %s", templateName))
+		}
+		return errors.Wrap(err, "Failed to retrieve DBC File")
+	}
+
+	// Return the DBC file itself
+	return c.Status(fiber.StatusOK).SendString(dbResult.DBCFile)
```

Comment:
let's set a header for the content-type of this response type being `text/plain`.
[Common mime types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types#)
[Fiber docs on setting headers](https://docs.gofiber.io/api/ctx#set)

Reply:
done